sandia_hand_left_controller:
  type: "robot_mechanism_controllers/JointTrajectoryActionController"
  joints:
    - left_f0_j0
    - left_f0_j1
    - left_f0_j2
    - left_f1_j0
    - left_f1_j1
    - left_f1_j2
    - left_f2_j0
    - left_f2_j1
    - left_f2_j2
    - left_f3_j0
    - left_f3_j1
    - left_f3_j2 
  gains:
    left_f0_j0 : {p: 1.0, d: 0, i: 0, i_clamp: 0}
    left_f0_j1 : {p: 1.0, d: 0, i: 0, i_clamp: 0}
    left_f0_j2 : {p: 1.0, d: 0, i: 0, i_clamp: 0}
    left_f1_j0 : {p: 1.0, d: 0, i: 0, i_clamp: 0}
    left_f1_j1 : {p: 1.0, d: 0, i: 0, i_clamp: 0}
    left_f1_j2 : {p: 1.0, d: 0, i: 0, i_clamp: 0}
    left_f2_j0 : {p: 1.0, d: 0, i: 0, i_clamp: 0}
    left_f2_j1 : {p: 1.0, d: 0, i: 0, i_clamp: 0}
    left_f2_j2 : {p: 1.0, d: 0, i: 0, i_clamp: 0}
    left_f3_j0 : {p: 1.0, d: 0, i: 0, i_clamp: 0}
    left_f3_j1 : {p: 1.0, d: 0, i: 0, i_clamp: 0}
    left_f3_j2 : {p: 1.0, d: 0, i: 0, i_clamp: 0}
  joint_trajectory_action_node:
    joints:
      - left_f0_j0
      - left_f0_j1
      - left_f0_j2
      - left_f1_j0
      - left_f1_j1
      - left_f1_j2
      - left_f2_j0
      - left_f2_j1
      - left_f2_j2
      - left_f3_j0
      - left_f3_j1
      - left_f3_j2 
    constraints:
      goal_time: 0.6
      left_f0_j0:
        goal: 0.02
      left_f0_j1:
        goal: 0.02
      left_f0_j2:
        goal: 0.02
      left_f1_j0:
        goal: 0.02
      left_f1_j1:
        goal: 0.02
      left_f1_j2:
        goal: 0.02
      left_f2_j0:
        goal: 0.02
      left_f2_j1:
        goal:0.02
      left_f2_j2:
        goal: 0.02
      left_f3_j0:
        goal: 0.02
      left_f3_j1:
        goal: 0.02
      left_f3_j2:
        goal: 0.02
